packages/block-editor/src/components/inserter/media-tab/hooks.js

```diff
@@ -98,13 +98,6 @@ function useInserterMediaCategories() {
 			) {
 				return false;
 			}
-			// When a category has set `isExternalResource` to `true`, we
-			// don't need to check for allowed mime types, as they are used
-			// for restricting uploads for this media type and not for
-			// inserting media from external sources.
-			if ( category.isExternalResource ) {
```

Comment:
Now we upload the images from external sources, so we have to check the `allowedMimeTypes` setting.